src/main/fc/stats.c

```diff
@@ -55,6 +62,7 @@ void statsOnDisarm(void)
     if (statsConfig()->stats_enabled) {
         uint32_t dt = (millis() - arm_millis) / 1000;
         if (dt >= MIN_FLIGHT_TIME_TO_RECORD_STATS_S) {
+            statsConfigMutable()->stats_flight_count = prev_flight_count + 1; // only increment once per power on
```

Comment:
This would just need to be `statsConfigMutable()->stats_flight_count++;`

Reply:
If we do it like you said we'd get a flight count increment each time a craft is disarmed and armed; by the definition of a flight that I'm going by we should only get one increment per battery pack (if it was flown longer than MIN_FLIGHT_TIME_TO_RECORD_STATS_S). As far as I can tell that's the most commonly used definition for a flight, and necessitates the function and variable I added. For reference, that's the heuristic behind a popular plugin for flight counting by Offer Shmuely https://github.com/offer-shmuely/edgetx-x10-widgets/tree/master.

Reply:
I'm not convinced that is the best way to define a flight. We can arm and disarm on a workbench, then remove the battery. That would then be classed as a flight. 

We can do things better than EdgeTX. We know that we have moved. Maybe a threshold of x metres away from home would be acceptable to say this is a legitimate flight. 

But I do see the point of keeping the previous count 👍🏻

Reply:
That is a fair point, my goal was to allow flight counting even for simple models that only use inav for stab w/o a gps, but I can see that it'd be more foolproof with a lapsed distance check as you suggested while abandoning a small % of userbase.
The problem you described does apply to power used as well though, doesn't it? Arm on workbench, throttle for longer then the MIN_FLIGHT_TIME, disarm, used mah gets added to stats. Just a thought.
What do you think an agreeable distance threshold would be, eg 10m?

Reply:
There is already code to detect if the craft is flying ... related to the landing detection stuff. This could be used to make this more reliable if a GPS is available. If a GPS isn't used then just use Arming as the trigger instead.

Reply:
@breadoven, I researched a bit and came up with three potential checks: if craft flew a min distance (using getTotalTravelDistance()), if max distance from home was greater than some set minimum, or isLandingDetected() on disarm. The first two need a threshold value so I'm leaning to the isLanding check, is that what you had in mind?
On another topic, how do we check for GPS availability?

Reply:
There's `isFlightDetected()` in `navigation.c` that is used to detect when flight has started. It's the best way of knowing if a fixed wing is actually flying. For multirotor the in flight checks are limited but it's as good as you're going to get given a multirotor flight could just be hovering slightly above the ground.

You just need to remember `isFlightDetected` will be true at some point in the flight, usually immediately after launch, but now necessarily throughout the whole flight.

GPS availability ... `sensors(SENSOR_GPS)`.

Reply:
@breadoven forgot to tag you, pls check out my last comment :)

Reply:
Quick question, isn't there an odometer already in INAV? Why not just check if there's a change in that number?

Reply:
> Quick question, isn't there an odometer already in INAV? Why not just check if there's a change in that number? Just thinking out loud.

Yea that's more or less what I was thinking, maybe with a threshold set so that moving a meter for example doesn't register as a flight